### PR TITLE
[Function-Rewriter] Support BiasSplitGelu operator in `stable_diffusion_unet`

### PR DESCRIPTION
**Pull request by titaiwangms**
1. Support BiasSplitGelu in diffuser models
2. Add `function_unittest_producer.py` from onnx-rewriter

After(left) vs Before(right)

Screenshot 2024-04-18 102455

### CODE REVIEW

File: onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        op = self.onnx_opset
+        msft_opset = onnxscript.values.Opset("com.microsoft", 1)
+
+        def ggelu(input, weight, bias):
+            weight_transpose = op.Transpose(weight, [1, 0])
+            matmul_input = op.MatMul(input, weight_transpose)
+            return msft_opset.BiasSplitGelu(matmul_input, bias)
+
+        function_proto = onnxscript.script(default_opset=op)(ggelu).to_function_proto()  # type: ignore[arg-type]
+        return function_proto, (onnx.helper.make_operatorsetid("com.microsoft", 1),)
```
Comment (justinchuby) [inline onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py:32]: Does this proto return need to change after the integration?

Comment (titaiwangms) [reply onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py:32]: Function rewriter is something else, unrelated to pattern rewriter.

**Comment by titaiwangms** [comment]
@justinchuby @gramalingam PTAL

File: onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import onnx
+
+import onnxscript
+from onnxscript.rewriter import function_rule
+
+logger = logging.getLogger(__name__)
+
+
+class GegluRewriteRule(function_rule.FunctionRewriteRule):
```
Comment (gramalingam) [inline onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py:13]: Is this a typo? Is it meant to be `Gelu...`?

Comment (gramalingam) [reply onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py:13]: Same below

Comment (titaiwangms) [reply onnxscript/rewriter/onnxruntime/transformers/biassplitgelu.py:13]: It's an activation function in diffuser: 
https://github.com/huggingface/diffusers/blob/e23c27e90524c3c3e93525d03f18408b84372012/src/diffusers/models/activations.py#L77

File: onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+from onnxscript import _legacy_ir as ir
+from onnxscript._legacy_ir import visitor
+from onnxscript.utils import evaluation_utils, utils
+
+logger = logging.getLogger(__name__)
+
+
+# Copied from common.py from pytorch torchbench
+def save_tensor_data(numpy_tensor, output_path: str):
+    from onnx import numpy_helper
```
Comment (justinchuby) [inline onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py:34]: Move import to top?

File: onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py
Comment (justinchuby) [inline onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py:1]: Should this be moved to the top level tools/ ?

**Review by justinchuby — APPROVED** [review]

File: onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py
Comment (titaiwangms) [reply onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py:1]: This only serves function rewriter to create unittests (onnxscript/rewriter/onnxruntime/transformers). What do you think?

Comment (justinchuby) [reply onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py:1]: I suggest moving it to tools/rewriter. Only importable modules and their unit tests should live in the package directory.

Comment (titaiwangms) [reply onnxscript/rewriter/onnxruntime/transformers/tools/function_unittest_producer.py:1]: Done

**Pull request merged**